app/controllers/application_controller.rb

```diff
@@ -3,6 +3,11 @@ class ApplicationController < ActionController::API
   include Pundit
 
   # child must define authenticate method
+  before_action :register_services
   before_action :authenticate
   after_action :verify_authorized
+
+  def register_services
+    @services = ServiceContainer.new
```

Comment:
I guess doing this means all services are made available to all controllers and the dependency injection is more implicit. What you think of this approach here? http://gmmoreira.com/2017-08-20/using-dry-system-in-a-rails-project.html

Reply:
Yeah I wasn't sure of the _best_ way of doing this, though I'm not a fan of the approach that the article takes as it doesn't mention DI of services which depend on others.

I might remove this from the `application_controller` for now (i didn't intend to commit it originally) and keep it in the courses controller which is where it's used and see what we evolve into. It might just be we make a concern in the future that registers the container, but I think we can defer the decision.